ctlog.go

```diff
@@ -217,3 +224,38 @@ var stringToKeyUsage = map[string]x509.ExtKeyUsage{
 	"MicrosoftServerGatedCrypto": x509.ExtKeyUsageMicrosoftServerGatedCrypto,
 	"NetscapeServerGatedCrypto":  x509.ExtKeyUsageNetscapeServerGatedCrypto,
 }
+
+// NewCTHTTPServer creates a Tessera based CT log pluged into an HTTP server.
+// The HTTP server handlers implement https://c2sp.org/static-ct-api write
+// endpoints.
+// TODO(phboneff): see if we can return an HTTP handler
+func NewCTHTTPServer(ctx context.Context, origin string, signer crypto.Signer, cfg ChainValidationConfig, cs CreateStorage, httpDeadline time.Duration, maskInternalErrors bool) (*http.ServeMux, error) {
+	log, err := newLog(ctx, origin, signer, cfg, cs)
+	if err != nil {
+		return nil, fmt.Errorf("newLog(): %v", err)
+	}
+
+	opts := &HandlerOptions{
+		Deadline:           httpDeadline,
+		RequestLog:         &DefaultRequestLog{},
+		MaskInternalErrors: maskInternalErrors,
+		TimeSource:         sysTimeSource,
+	}
+
+	handlers := NewPathHandlers(opts, log)
+
+	klog.Info("**** CT HTTP Server Starting ****")
+	// Allow cross-origin requests to all handlers registered on corsMux.
+	// This is safe for CT log handlers because the log is public and
+	// unauthenticated so cross-site scripting attacks are not a concern.
+	corsMux := http.NewServeMux()
+	corsHandler := cors.AllowAll().Handler(corsMux)
+	http.Handle("/", corsHandler)
```

Comment:
This feels a little scary as-is since we don't know what other handlers _could_ be registered by the caller on this same mux, but I _think_ the change above to return just a `http.Handler` would help to avoid that particular danger.

Reply:
I had left a TODO to come back to this later, but I've added a few things in this PR to remove a bunch of stuff that made this hard to follow. PTAL.